source/common/event/libevent_scheduler.h

```diff
@@ -41,22 +44,32 @@ class LibeventScheduler : public Scheduler {
    */
   event_base& base() { return *libevent_; }
 
+  /**
+   * Register callback to be called in the event loop prior to polling for
+   * events. Must not be called more than once. |callback| must not be null.
```

Comment:
Matt: In some other codebases I have been working with it is customary to enforce API contract with DCHECKs.  In this case it would look like this:

DCHECK(callback);
DCHECK(!callback_);

Is this customary in Envoy?

Reply:
I was about to ask for this. :) Yes please (using ASSERT)